src/generators.jl

```diff
@@ -814,3 +814,88 @@ Usage
 `A = forest_fire_graph(c, k, p)` returns a forest-fire sample. 
 `A = forest_fire_graph(A0, k, p)` 
 """
+
+
+"""
+`partial_duplication`
+=====================
+
+A random graph model based off of the evolution of protein-protein interaction
+networks. The method takes an undirected graph and runs a number of steps 
+which randomly selects an existing vertex, duplicates the node, and keeps the
+edges between its neighbor with a given probability. In the original work 
+the authors prove that new vertices will tend to become isolated for 
+p <= 0.567143. 
+
+ src: https://arxiv.org/pdf/1408.0904.pdf
+
+Input
+-----
+- 'A::MatrixNetwork{T}': seed matrix network 
+- `steps::Int': The number of steps to run the procedure
+- `new_edge_p::Float64': the probability 
+
+Output
+------
+- a new matrix network generated through the partial duplication procedure. 
```

Comment:
Clarify if it's undirected or directed given undirected/directed input, respectively (symmetric, non-symmetric)